L1Trigger/L1TGlobal/src/AXOL1TLCondition.cc

```diff
@@ -0,0 +1,254 @@
+      if ((candEtSumVec->at(useBx, iEtSum))->getType() == l1t::EtSum::EtSumType::kMissingEt) {
+        EtSumInput[0] =
+            ((candEtSumVec->at(useBx, iEtSum))->hwPt()) / 2;  //have to do hwPt/2 in order to match original et inputs
+        // EtSumInput[1] = (candEtSumVec->at(useBx, iEtSum))->hwEta(); //this one is zero, so leave it zero
+        EtSumInput[2] = (candEtSumVec->at(useBx, iEtSum))->hwPhi();
+      }
+    }
+  }
+
+  //next egammas
+  if (NCandEG > 0) {  //check if not empty
+    for (int iEG = 0; iEG < NCandEG; iEG++) {
+      if (iEG < NEgammas) {  //stop if fill the Nobjects we need
+        EgammaInput[0 + (3 * iEG)] = ((candEGVec->at(useBx, iEG))->hwPt()) /
+                                     2;  //index 0,3,6,9 //have to do hwPt/2 in order to match original et inputs
+        EgammaInput[1 + (3 * iEG)] = (candEGVec->at(useBx, iEG))->hwEta();  //index 1,4,7,10
+        EgammaInput[2 + (3 * iEG)] = (candEGVec->at(useBx, iEG))->hwPhi();  //index 2,5,8,11
+      }
+    }
+  }
+
+  //next muons
+  if (NCandMu > 0) {  //check if not empty
+    for (int iMu = 0; iMu < NCandMu; iMu++) {
+      if (iMu < NMuons) {  //stop if fill the Nobjects we need
+        MuInput[0 + (3 * iMu)] = ((candMuVec->at(useBx, iMu))->hwPt()) /
+                                 2;  //index 0,3,6,9 //have to do hwPt/2 in order to match original et inputs
+        MuInput[1 + (3 * iMu)] = (candMuVec->at(useBx, iMu))->hwEta();  //index 1,4,7,10
+        MuInput[2 + (3 * iMu)] = (candMuVec->at(useBx, iMu))->hwPhi();  //index 2,5,8,11
+      }
+    }
+  }
+
+  //next jets
+  if (NCandJet > 0) {  //check if not empty
+    for (int iJet = 0; iJet < NCandJet; iJet++) {
+      if (iJet < NJets) {  //stop if fill the Nobjects we need
+        JetInput[0 + (3 * iJet)] = ((candJetVec->at(useBx, iJet))->hwPt()) /
+                                   2;  //index 0,3,6,9...27 //have to do hwPt/2 in order to match original et inputs
+        JetInput[1 + (3 * iJet)] = (candJetVec->at(useBx, iJet))->hwEta();  //index 1,4,7,10...28
+        JetInput[2 + (3 * iJet)] = (candJetVec->at(useBx, iJet))->hwPhi();  //index 2,5,8,11...29
+      }
+    }
+  }
+
+  //now put it all together-> EtSum+EGamma+Muon+Jet into ADModelInput
+  int index = 0;
+  for (int idET = 0; idET < EtSumVecSize; idET++) {
+    ADModelInput[index++] = EtSumInput[idET];
+  }
+  for (int idEG = 0; idEG < EGVecSize; idEG++) {
+    ADModelInput[index++] = EgammaInput[idEG];
+  }
+  for (int idMu = 0; idMu < MuVecSize; idMu++) {
+    ADModelInput[index++] = MuInput[idMu];
+  }
+  for (int idJ = 0; idJ < JVecSize; idJ++) {
+    ADModelInput[index++] = JetInput[idJ];
+  }
+
+  //now run the inference
+  model->prepare_input(ADModelInput);  //scaling internal here
+  model->predict();
+  model->read_result(&ADModelResult);  // this should be the square sum model result
+
+  result = ADModelResult.first;
+  loss = ADModelResult.second;
+  score = ((loss).to_float()) * 16.0;  //scaling to match threshold
```

Comment:
What is this `16.0` here?

Reply:
This is a scaling to make the output loss match the threshold values defined for the test crate

Reply:
Could you please create a const with a name that might indicate that and use that instead or mention the test crate in the comment. It's just a picky maintenance/clarity of code thing.